suseconnect/migration.go

```diff
@@ -150,41 +147,36 @@ func migrationMain() {
 	}
 
 	if selfUpdate {
+		// reset root (if set) as the update stack can be outside of
+		// the to be updated system
+		connect.CFG.FsRoot = ""
 		echo := connect.SetSystemEcho(true)
-		//   # Update stack can be outside of the to be updated system
-		//   cmd = "zypper " +
-		//         (options[:non_interactive] ? "--non-interactive " : "") +
-		//         (options[:verbose] ? "--verbose " : "") +
-		//         (options[:quiet] ? "--quiet " : "") +
-		//         "patch-check --updatestack-only"
-		//   print "\nExecuting '#{cmd}'\n\n" unless options[:quiet]
-		//   if !system cmd
-		//     if $?.exitstatus >= 100
-		//       # install pending updates and restart
-		//       cmd = "zypper " +
-		//             (options[:non_interactive] ? "--non-interactive " : "") +
-		//             (options[:verbose] ? "--verbose " : "") +
-		//             (options[:quiet] ? "--quiet " : "") +
-		//             "--no-refresh patch --updatestack-only"
-		//       print "\nExecuting '#{cmd}'\n\n" unless options[:quiet]
-		//       system cmd
-
-		//       # stop infinite restarting
-		//       # check that the patches were really installed
-		//       cmd = "zypper " +
-		//         (options[:root] ? "--root #{options[:root]} " : "") +
-		//         "--non-interactive --quiet --no-refresh patch-check --updatestack-only > /dev/null"
-		//       if ! system cmd
-		//         print "patch failed, exiting.\n"
-		//         exit 1
-		//       end
-
-		//       print "\nRestarting the migration script...\n" unless options[:quiet]
-		//       exec $0, *save_argv
-		//     end
-		//     exit 1
-		//   end
+		if pending, err := connect.PatchCheck(true, quiet, verbose, nonInteractive, false); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		} else if pending {
+			// install pending updates and restart
+			// TODO: check error?
+			connect.Patch(true, quiet, verbose, nonInteractive, true)
+			// stop infinite restarting
+			// check that the patches were really installed
+			if pending, err := connect.PatchCheck(true, true, false, true, true); pending || err != nil {
+				fmt.Println("patch failed, exiting.")
```

Comment:
Maybe `Patch self-update failed, exiting` and add the `err` if any

Reply:
but as it stands, it does not say why it failed. Were there still pending updates, or was there an error, and if there was an error what was it.